test/apps/iodemo/io_demo.cc

```diff
@@ -109,6 +107,13 @@ template<class BufferType, bool use_offcache = false> class ObjectPool {
         }
     }
 
+    void init()
+    {
+        for (size_t i = 0; i < _offcache; ++i) {
+            _offcache_queue.push(get_free());
+        }
+    }
```

Comment:
can we do it in the constructor?

Reply:
I am afraid we can't. get_free() calls construct() which is a virtual function. That is the reason why I move these codes into a newly created init().

Reply:
ok, so WDYT about these adjustments:
1. pass "num_offcache" parameter to init() instead of the constructor. so no need to save a member variable _offcache.
2. in get(), remove "_offcache_queue.empty()" condition - to optimize (since now offcache is fixed)

Reply:
1. I thought it before. Currently offcache is passed in as a parameter to ObjectPool's constructor and children classes MemoryPool and BufferMemoryPool also pass it to upper constructor. Removing offcache from the parameter list of ObjectPool's constructor would incur more places to be modified. Do you really want me do this way?
2. I am a bit lost here. Since we don't have _offcache member variable in ObjectPool, _offcache_queue.empty() is used to check if _offcache is zero or not.  By removing "_offcache_queue.empty()" condition use_offcache alone seems make no sense to me. Am I missing something?

Reply:
1. modifying more code by removing is ok
2. why use_offcache (template param) is not enough to check off-cache condition?

Reply:
1. OK. I will do it in next commit.
2. use_offcache == true is not mutual exclusive with offcache == 0 so both conditions need to be checked. Do you want to add such restriction? That way seems make things complicated

Reply:
2. ok